diff -rupN opensim-0.8.2.1-source/OpenSim/Region/Framework/Scenes/ScenePresence.cs new/OpenSim/Region/Framework/Scenes/ScenePresence.cs
--- opensim-0.8.2.1-source/OpenSim/Region/Framework/Scenes/ScenePresence.cs	Wed Dec 09 21:14:28 2015
+++ new/OpenSim/Region/Framework/Scenes/ScenePresence.cs	Tue Oct 17 11:23:41 2017
@@ -2543,7 +2543,7 @@ namespace OpenSim.Region.Framework.Scene
             if (pos.Z - terrainHeight < 0.2)
                 pos.Z = terrainHeight;
 
-            if (noFly)
+            if (noFly || !landAtTarget)
                 Flying = false;
             else if (pos.Z > terrainHeight)
                 Flying = true;
