--- opensim-0.8.2.1-source/OpenSim/Region/Framework/Scenes/SceneObjectGroup.cs	Wed Dec 09 20:14:28 2015
+++ opensim-0.8.2.1-source-fix/OpenSim/Region/Framework/Scenes/SceneObjectGroup.cs	Sat Mar 24 13:29:34 2018
@@ -1906,7 +1906,7 @@ namespace OpenSim.Region.Framework.Scene
             {
                 ScenePresence avatar = m_scene.GetScenePresence(AttachedAvatar);
 
-                if (avatar != null)
+                if (avatar != null && !avatar.IsSatOnObject)
                     avatar.MoveToTarget(target, false, false);
             }
             else
