--- opensim-0.9.0.0-source/OpenSim/Region/Framework/Scenes/SceneObjectGroup.cs	Tue Aug 15 18:31:06 2017
+++ opensim-0.9.0.0-source-fix/OpenSim/Region/Framework/Scenes/SceneObjectGroup.cs	Sat Mar 24 13:30:44 2018
@@ -2789,7 +2789,7 @@ namespace OpenSim.Region.Framework.Scene
             {
                 ScenePresence avatar = m_scene.GetScenePresence(AttachedAvatar);
 
-                if (avatar != null)
+                if (avatar != null && !avatar.IsSatOnObject)
                     avatar.MoveToTarget(target, false, false);
             }
             else
